packages/@react-aria/datepicker/src/useDateField.ts

```diff
@@ -64,8 +64,23 @@ export function useDateField<T extends DateValue>(props: AriaDateFieldProps<T>,
   });
 
   let {focusWithinProps} = useFocusWithin({
-    onBlurWithin() {
+    ...props,
+    onBlurWithin: (e: FocusEvent) => {
       state.confirmPlaceholder();
+
+      if (props.onBlur) {
+        props.onBlur(e);
+      }
+    },
+    onFocusWithin: (e: FocusEvent) => {
+      if (props.onFocus) {
+        props.onFocus(e);
+      }
+    },
```

Comment:
You probably don't need these wrapper functions. You could do something like this:

```js
onFocusWithin: props.onFocus
```